### DISCUSSION TITLE
config/yml file configuration

### DISCUSSION THREAD

**Discussion by ntsakoulis** [fastmachinelearning/hls4ml#929] [General]
hello everyone.

I have some questions about the yml file configuration. any help will be wellcomed.

 # Model section (Keras model)
1 KerasJson: keras/KERAS_3layer.json
2 KerasH5:   keras/KERAS_3layer_weights.h5 #You can also use h5 file from Keras's model.save() without supplying json file.
3 InputData: keras/KERAS_3layer_input_features.dat
4 OutputPredictions: keras/KERAS_3layer_predictions.dat

at this point of yml file. i save my keras model as .h5 file. so i remove first line and i only keep the second ? 
also where is the decalration golden_output.dat file??

also
HLSConfig:
  Model:
    Precision: ap_fixed<16,6>
    ReuseFactor: 1
    Strategy: Latency
  LayerType:
    Dense:
      ReuseFactor: 2
      Strategy: Resource
    

i have understand this section of code. but where can i find available cionfiguration options ?? is there any doc ? i can not find anything 

Last, i built a yml file and from cli i typed    hl4ml convert -p project_name and the folder that was created did not include any vivado project in order to open it and make any work directly from vivado. Should i use only python api for hls4ml??

thnaks for your time ,

**Comment by vloncar** [comment]
Use the python API, the CLI is deprecated. Did you try reading the [documentation](https://fastmachinelearning.org/hls4ml/)?

Out of curiosity, where did you get advice to start with the CLI instead of the API that is suggested on the project page and in all documentation?

**Reply by ntsakoulis** [reply]
from google, but it does not matter. I'll make use of python API. Of course i tried the documentation but i can not find answers in my questions Vladimir. Can you help me with my questions?
Thanks,

**Reply by cheese-leopard** [reply]
https://github.com/fastmachinelearning/hls4ml-tutorial/blob/main/part1_getting_started.ipynb
part1 to 7 may be able to figure out ur questions
